src/TTrigger.cpp

```diff
@@ -355,10 +356,14 @@ bool TTrigger::match_perl(char* subject, const QString& toMatch, int regexNumber
         char* tabptr = reinterpret_cast<char*>(name_table);
         for (i = 0; i < namecount; i++) {
             int n = (tabptr[0] << 8) | tabptr[1];
-            auto name = QString::fromUtf8( tabptr + 2, name_entry_size - 3).trimmed();
-            auto capture = QString::fromUtf8(subject + ovector[2*n], ovector[2*n+1] - ovector[2*n]);
+            auto name = QString::fromUtf8(tabptr + 2, name_entry_size - 3).trimmed().remove(QStringLiteral("\u0000"));
```

Comment:
Can you add a comment for the `remove()`? It's purpose not immediately clear.

Reply:
Sure, just added.
it does not propagate to Lua, but I've noticed that this null is somehow added sometimes to name of capture group.
Might be better way to deal with it - but I don't know it - this for sue helps.

Reply:
You are dealing with data from a C library and it's outputs are going to be `\0` terminated `const char[]` C strings. The `name_entry_size` value is going to be a strìng length argument and if you are trying to return partial string extraction with the original strings you should be referring to it (*I* believe) with a `&tabptr[2]` rather than `tabptr + 2`. Indeed given that as a first argument to the `QString::fromUtf8(...)` method I believe it is smart enough to remove just the first 2 characters from whatever it is that the original `tabptr` pointed to without having to specify an explicit length of string to consider as a second argument (and thereby not needing the `QString::remove(...)` either...

Reply:
`remove` remark - I've already managed to figure that one out, by removing second argument all was fixed and `remove` could be deleted.
I've also just changed first argument as suggested.